From 6c60a2104eeebd96a6037e565fcd34d8803be938 Mon Sep 17 00:00:00 2001
From: Harish Mahendrakar <harish.mahendrakar@ittiam.com>
Date: Fri, 18 Nov 2016 14:23:34 +0530
Subject: [PATCH 1/3] Fixed handling invalid chroma tu size for error clips

Limit func_idx to valid range to ensure invalid functions are not
called when wrong TU size is signalled for chroma due to error in
parsing

Bug: 32915871
Change-Id: I662212eb2e9b8994e7e85780e667f14df73b5905
(cherry picked from commit a76773ab749bd57f3467c79aa60c16c1f2c87380)
---
 decoder/ihevcd_iquant_itrans_recon_ctb.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/decoder/ihevcd_iquant_itrans_recon_ctb.c b/decoder/ihevcd_iquant_itrans_recon_ctb.c
index a558644..f6c89b7 100644
--- a/decoder/ihevcd_iquant_itrans_recon_ctb.c
+++ b/decoder/ihevcd_iquant_itrans_recon_ctb.c
@@ -884,6 +884,11 @@ WORD32 ihevcd_iquant_itrans_recon_ctb(process_ctxt_t *ps_proc)
                     src_strd = trans_size;
 
                     func_idx = 1 + 4 + log2_uv_trans_size_minus_2; /* DST func + Y funcs + cur func index*/
+
+                    /* Handle error cases where 64x64 TU is signalled which results in 32x32 chroma.
+                     * By limiting func_idx to 7, max of 16x16 chroma is called */
+                    func_idx = MIN(func_idx, 7);
+
                     e_trans_type = (TRANSFORM_TYPE)(log2_uv_trans_size_minus_2 + 1);
                     /* QP for U */
                     i1_chroma_pic_qp_offset = ps_pps->i1_pic_cb_qp_offset;
-- 
2.7.4

